src/main/java/mate/jdbc/dao/ManafacturerDaoImpl.java

```diff
@@ -0,0 +1,113 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionDb;
+
+@Dao
+public class ManafacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertRequest = "INSERT INTO manufacturers(name, country)"
+                + " values(?,?)";
+        try (Connection connection = ConnectionDb.getConnection();
+                PreparedStatement createStatement
+                        = connection.prepareStatement(insertRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            createStatement.setString(1, manufacturer.getName());
+            createStatement.setString(2, manufacturer.getCountry());
+            createStatement.executeUpdate();
+            ResultSet generatedKeys = createStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not create manufacturer" + manufacturer, e);
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getRequest = "SELECT * FROM manufacturers WHERE id = ?";
```

Comment:
```suggestion
        String getRequest = "SELECT * FROM manufacturers WHERE id = ? AND is_deleted = FALSE;";
```

[Checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-JDBC/jdbc-intro/JDBC-intro_checklist.html)
Remember that we can’t do next operations on deleted manufacturers: get(), getAll(), update().